documentation/source/testbench_tools.rst

```diff
@@ -0,0 +1,189 @@
+                self.dut = dut
+                self.stream_in = AvalonSTDriver(dut, "stream_in", dut.clk)
+                self.stream_in_recovered = AvalonSTMonitor(dut, "stream_in", dut.clk,
+                                                           callback=self.model)
+                
+                # Set verbosity on our various interfaces
+                level = logging.DEBUG if debug else logging.WARNING
+                self.stream_in.log.setLevel(level)
+                self.stream_in_recovered.log.setLevel(level)
+                self.dut.reset_n._log(setLevel(logging.DEBUG)
+
+And when the logging is actually called
+
+.. code-block:: python
+
+        class AvalonSTPkts(BusMonitor):
+        ...
+            @coroutine
+            def _monitor_recv(self):
+                ...
+                self.log.info("Received a packet of %d bytes" % len(pkt))
+
+        class Scoreboard(object):
+            ...
+            def add_interface(self):
+                ...
+                self.log.info("Created with reorder_depth %d" % reorder_depth)
+
+        class EndianSwapTB(object):
+            ...
+            @cocotb.coroutine
+            def reset():
+                self.dut._log.debug("Resetting DUT")
+                
+
+will display as something like
+
+.. code-block:: bash
+
+    0.00ns INFO                   cocotb.scoreboard.endian_swapper_sv       scoreboard.py:177  in add_interface                   Created with reorder_depth 0    
+    0.00ns DEBUG                  cocotb.endian_swapper_sv           .._endian_swapper.py:106  in reset                           Resetting DUT
+    160000000000000.00ns INFO     cocotb.endian_swapper_sv.stream_out           avalon.py:151  in _monitor_recv                   Received a packet of 125 bytes
+
+                
+Busses
+======
+
+Busses are simply defined as collection of signals. The :py:class:`Bus` class will automatically bundle any group of signals together that are named similar to dut.<bus_name><separator><signal_name>. for instance,
+
+.. code-block:: python
+
+        dut.stream_in_valid
+        dut.stream_in_data
+        
+have a bus name of ``stream_in``, a separator of ``_``, and signal names of ``valid`` and ``data``. A list of signal names, or a dictionary mapping attribute names to signal names is also passed into the :py:class:`Bus` class. Busses can have values driven onto them, be captured (returning a dictionary), or sampled and stored into a similar object. 
+
+.. code-block:: python
+
+                stream_in_bus = Bus(dut, "stream_in", ["valid", "data"]) # '_' is the default separator
+
+
+Driving Busses
+==============
+
+examples and specific bus implementation bus drivers (amba, avalon, xgmii, and others) exist in the :py:class:`Driver` class enabling a test to append transactions to perform the serialization of transactions onto a physical interface. Here's an example using the avalon bus driver in the endian swapper example
```

Comment:
"Examples" (capital E)

Reply:
That's a much clearer description for <=. I'll use that. 

I guess I missed the errant (, and I fixed the missing E above.